File: code_samples/scala/networking_websockets.scala
/**
 * Networking: Websockets
 * AI/ML Training Sample
 */

class Websockets {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object WebsocketsExample {
  def main(args: Array[String]): Unit = {
    val instance = new Websockets()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/r/networking_websockets.r
# Networking: Websockets
# AI/ML Training Sample

Websockets <- setRefClass(
    "Websockets",
    fields = list(data = "character"),
    methods = list(
        initialize = function() {
            data <<- ""
        },
        process = function(input) {
            data <<- input
        },
        getData = function() {
            return(data)
        },
        validate = function() {
            return(nchar(data) > 0)
        }
    )
)

# Example usage
instance <- Websockets$new()
instance$process("example")
cat("Data:", instance$getData(), "\n")
cat("Valid:", instance$validate(), "\n")
